tests/legacy-cli/e2e/tests/update/update-10.ts

```diff
@@ -0,0 +1,62 @@
+import { SemVer } from 'semver';
+import { createProjectFromAsset } from '../../utils/assets';
+import { expectFileMatchToExist, readFile } from '../../utils/fs';
+import { setRegistry } from '../../utils/packages';
+import { ng, noSilentNg } from '../../utils/process';
+import { isPrereleaseCli, useCIChrome, useCIDefaults, NgCLIVersion } from '../../utils/project';
+
+export default async function () {
+  try {
+    // We need to use the public registry because in the local NPM server we don't have
+    // older versions @angular/cli packages which would cause `npm install` during `ng update` to fail.
+    await setRegistry(false);
+    await createProjectFromAsset('10.0-project', true);
+
+    // CLI proiject version
+    const { version: cliVersion } = JSON.parse(
+      await readFile('./node_modules/@angular/cli/package.json'),
+    );
+    const cliMajorProjectVersion = new SemVer(cliVersion).major;
+
+    // CLI current version.
+    const cliMajorVersion = NgCLIVersion.major;
+
+    for (let version = cliMajorProjectVersion + 1; version < cliMajorVersion; version++) {
+      // Run all the migrations until the current build major version - 1.
+      // Example: when the project is using CLI version 10 and the build CLI version is 14.
+      // We will run the following migrations:
+      // - 10 -> 11
+      // - 11 -> 12
+      // - 12 -> 13
+      const { stdout } = await ng('update', `@angular/cli@${version}`, `@angular/core@${version}`);
+      if (!stdout.includes("Executing migrations of package '@angular/cli'")) {
+        throw new Error('Update did not execute migrations. OUTPUT: \n' + stdout);
+      }
+    }
+  } finally {
+    await setRegistry(true);
+  }
+
+  // Update Angular current build
+  const extraUpdateArgs = isPrereleaseCli() ? ['--next', '--force'] : [];
+  // For the latest/next release we purposely don't add `@angular/core`.
+  // This is due to our bumping strategy, which causes a period were `@angular/cli@latest` (v12.0.0) `@angular/core@latest` (v11.2.x)
```

Comment:
I don't fully understand this. Does this happen right after we release a major version but still have dev dependencies on FW's latest patch? If we skip `@angular/core` here, will we ever run update schematics for the upcoming major?

Reply:
There can be a period when we bump the master to a new major ex: 14.0.0-next.0 but the same version of the FW is not yet available on NPM. This would cause the test to fail.

Reply:
`ng update @angular/cli` will always update and run `@angular/core` because the later is a peer dependency of the CLI.

Reply:
I think major releases do an immediate prerelease specifically to avoid that state. The [release checklist](https://docs.google.com/document/d/1GAyhWTTxsb28AlJZpn4yjrDZW3nT8NMv6xP5aQ9qcvA/edit?resourcekey=0-eM9Rx3k1fFjNaoTbcKEyhQ#) has FW release the major version *and* the next before CLI releases it's major and bumps to next. Does that mean this state shouldn't happen?

Reply:
I requested access to the document.

Ill dig more into this, but I remember it was a recurring issue, and this was a workaround without any side-effect.

Reply:
The problem was mainly during the RC periods, when FW do not release master as next version.

Ex: CLI master is 14.0.0-next.0, but @angular/core@next is pointing to 13.2.0-rc.0 and on NPM @angular/core@14.0.0-next.0 doesn’t exist yet. This would cause `ng update @angular/core@14` to fail.

Reply:
Gave you access, I assumed the team already had it. Agreed that we should chat more on Monday.

Reply:
I updated the comment to capture as much as I can to why this needs to be here.

Reply:
Thanks for the comment, I hadn't considered the case where FW ships a major version as stable but hasn't yet tagged it as `latest`. That addresses my concern, and installing just `@angular/cli` is a reasonable workaround.